src/avatar.js

```diff
@@ -111,7 +111,7 @@ function createAvatarComponent({ sources = [] }) {
         componentWillUnmount() {
             this.mounted = false;
             if (this.state.internal) {
-                this.state.internal.active = false;
+                this.setState({ internal: { active: false }});
```

Comment:
This will break, `internal` needs to be the exact same object as there are places that perform a `===` comparison to ensure the component is only updated while mounted an active.

https://github.com/Sitebase/react-avatar/blob/2b1aa8cae7676ded1a23d7b27587d0f8d45b1370/src/internal-state.js#L10-L22